### PR TITLE
LPS-124828 Take into account language id when updading the editable values

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

**Comment by victorg1991** [comment]
ci:test:sf

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js [hunk 1]
```diff
@@ -59,18 +59,24 @@ const useCollectionConfig = () => {
...
 	const {className, classPK} = context.collectionItem || {};
 
+	const hasLocalizable =
+		fragmentEntryLink.configuration?.fieldSets?.some((fieldSet) =>
+			fieldSet.fields.some((field) => field.localizable)
+		) ?? false;
+
 	useEffect(() => {
-		if (context.collectionItemIndex != null) {
+		if (context.collectionItemIndex != null || hasLocalizable) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js:74]: Maybe `context.collectionItemIndex !======== null`

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js [hunk 2]
```diff
@@ -59,18 +59,24 @@ const useCollectionConfig = () => {
...
 };
 
-const useGetContent = (fragmentEntryLink, segmentsExperienceId) => {
+const useGetContent = (fragmentEntryLink, languageId, segmentsExperienceId) => {
 	const context = useContext(CollectionItemContext);
 	const dispatch = useDispatch();
 
 	const {className, classPK} = context.collectionItem || {};
 
+	const hasLocalizable =
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js:68]: Have you considered moving this calculation inside the useEffect hook depending on fieldSets?:

```
const fieldSets = fragmentEntryLink.configuration?.fieldSets;

useEffect(() => {
  // Bla bla
}, [fieldSets]);
```
I am not sure if fieldSets is changing a lot of times. If it is more or less static we can skip some loops for items inside collections.

**Review by p2kmgcl — COMMENTED** [review]
Just a typo and a small consideration :3
Awesome

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js (hunk 1 above)
Comment (victorg1991) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js:74]: I think it was like that to check both undefined and null

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js (hunk 2 above)
Comment (victorg1991) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js:68]: I'm assuming you meant `useMemo` right? I am going to switch that :p

Comment (victorg1991) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/CollectionItemContext.js]: Done!

**Comment by victorg1991** [comment]
ci:forward

**Comment by victorg1991** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
Hi @victorg1991, there is a regression issue.
**Step to reproduce:**
1. Add a new site
2. Add a web content based Basic Web Content
3. Type title, content, and summary > Publish
4. Add a page based on Blank
5. Add a Collection Display fragment to page
6. Select the Highest Rated Assets collection provider
7. Add the Content Display to the cell of Collection Display
8. Assert the title of web content is shown in Content Display
9. Click the body of Content Display > General tab
10. Select the Abstract in Template field

*Expected Results:*
The summary of web content should be shown in Content Display.

*Actual Results:*
The title of web content is still shown in Content Display.

**Comment by victorg1991** [comment]
Should be fixed now, thanks @Tim-Cao  :)

**Comment by victorg1991** [comment]
ci:forward

**Pull request closed without merging**